lib/s3routes/routesUtils.ts

```diff
@@ -317,9 +318,11 @@ function okContentHeadersResponse(
             overrideParams['response-content-encoding'];
     }
     setCommonResponseHeaders(addHeaders, response, log);
-    const httpCode = range ? 206 : 200;
-    log.debug('response http code', { httpCode });
-    response.writeHead(httpCode);
+    if (shouldWriteHeaders) {
```

Comment:
instead of adding a condition here, I would rather remove `writeHead()` from here, and move it to the one place where `okContentHeadersResponse` is called with the default value shouldWriterHeaders=true (i.e. in `responseContentHeaders()`)

Reply:
Done.